pkg/minikube/cluster/cluster.go

```diff
@@ -181,27 +181,124 @@ func StartCluster(h sshAble) error {
 	return nil
 }
 
-type fileToCopy struct {
+type CopyableFile interface {
+	io.Reader
+	GetLength() int
+	GetAssetName() string
+	GetTargetDir() string
+	GetTargetName() string
+	GetPermissions() string
+}
+
+type FileAsset struct {
+	Data        []byte
+	Reader      io.Reader
+	Length      int
 	AssetName   string
 	TargetDir   string
 	TargetName  string
 	Permissions string
 }
 
-var assets = []fileToCopy{
-	{
+func (f *FileAsset) Read(p []byte) (int, error) {
+	if f.Reader == nil {
+		contents, err := ioutil.ReadFile(f.AssetName)
+		if err != nil {
+			return 0, err
+		}
+		f.Data = contents
+		f.Length = len(contents)
+		f.Reader = bytes.NewReader(f.Data)
+	}
+	return f.Reader.Read(p)
+}
+
+// unsure if this should return special case int values or int, error
+func (f *FileAsset) GetLength() int {
+	return f.Length
+}
+
+func (f *FileAsset) GetAssetName() string {
+	return f.AssetName
+}
+
+func (f *FileAsset) GetTargetDir() string {
+	return f.TargetDir
+}
+
+func (f *FileAsset) GetTargetName() string {
+	return f.TargetName
+}
+
+func (f *FileAsset) GetPermissions() string {
+	return f.Permissions
+}
+
+type MemoryAsset struct {
+	Data        []byte
+	Reader      io.Reader
+	Length      int
+	AssetName   string
+	TargetDir   string
+	TargetName  string
+	Permissions string
+}
+
+func (m *MemoryAsset) Read(p []byte) (int, error) {
+	if m.Reader == nil {
```

Comment:
I would make this check if m.Data == nil, then you have a helper function called loadData which does this stuff. Then you can just do that same check here and in GetLength.